internal/webhooks/clusterclass.go

```diff
@@ -123,8 +143,86 @@ func (webhook *ClusterClass) validate(old, new *clusterv1.ClusterClass) error {
 	// Ensure spec changes are compatible.
 	allErrs = append(allErrs, check.ClusterClassesAreCompatible(old, new)...)
 
+	// Ensure no MachineDeploymentClass currently in use has been removed from the ClusterClass.
+	allErrs = append(allErrs, webhook.validateRemovedMachineDeploymentClassesAreNotUsed(old, new)...)
+
 	if len(allErrs) > 0 {
 		return apierrors.NewInvalid(clusterv1.GroupVersion.WithKind("ClusterClass").GroupKind(), new.Name, allErrs)
 	}
 	return nil
 }
+
+func (webhook *ClusterClass) validateRemovedMachineDeploymentClassesAreNotUsed(old, new *clusterv1.ClusterClass) field.ErrorList {
+	var allErrs field.ErrorList
+	if old == nil {
+		return allErrs
+	}
+	removedClasses := webhook.removedMachineClasses(old, new)
+
+	// If no classes have been removed return early as no further checks are needed.
+	if len(removedClasses) == 0 {
+		return allErrs
+	}
+
+	// Retrieve all clusters using the ClusterClass.
+	clusters, err := webhook.getClustersUsingClusterClass(old)
```

Comment:
In this case we will now have an apierror wrapped into an field error wrapped into an apierror. I'm not sure what the error for users will look like.

My intention was to suggest the following:
* return a regular error in `getClustersUsingClusterClass`
* ValidateDelete:
    * Take the error and wrap it into an apierror in `ValidateDelete`
* ValidateCreate/Update
   * Call `getClustersUsingClusterClass` directly in `validate` instead of in `validateRemovedMachineDeploymentClassesAreNotUsed`, if there is an error wrap it into an apierror and return it
        * If I understood it correctly we have to move `getClustersUsingClusterClass` that level up anyway sooner or later to avoid calling this func multiple times.

WDYT?

Reply:
I think we're going to have to wrap the client error into a field.ErrorList at that level anyway in order to retain the errors from earlier in validate. I'll update with what I have, but I don't see any direct and clean way to wrap all of these errors into something that will be perfect in every case.

Reply:
I'm not sure but when `getClustersUsingClusterClass` is called directly in `validate` I would have thought that we can just return the error we want, but maybe I'm missing something

Reply:
Validate will already have some errors wrapped in field.ErrorList by the time getClustersUsingClusterClass is called so we should wrap that error in the same way.

Reply:
As discussed, we now have one api error which contains a list of field errors and that's fine (as we still want to return the field errors)